File: tests/parsers/samples/code/go/simple.go
package samples

import "fmt"

type Metrics struct {
	Baseline float64
	RMS      float64
}

func RunSimple() Metrics {
	analyzer := NewComplexMetrics()
	return Metrics{
		Baseline: analyzer.Distance(10, 3),
		RMS:      analyzer.RootMeanSquare([]float64{1, 2, 3}),
	}
}

func Summarize() string {
	metrics := RunSimple()
	return fmt.Sprintf("baseline=%.2f, rms=%.2f", metrics.Baseline, metrics.RMS)
}

File: tests/parsers/samples/code/r/simple.r
run_simple <- function() {
  baseline <- complex_distance(10, 3, scale = 0.8)
  rms <- complex_root_mean_square(c(1, 2, 3))
  list(baseline = baseline, rms = rms)
}

summarize <- function() {
  metrics <- run_simple()
  sprintf('baseline=%.2f, rms=%.2f', metrics$baseline, metrics$rms)
}
